week1/part-2-warmup/src/main.rs

```diff
@@ -4,18 +4,38 @@ use std::collections::HashSet;
 
 fn main() {
     println!("Hi! Try running \"cargo test\" to run tests.");
+    let mut v = vec![3, 1, 0, 1, 4, 4];
+    let _r = add_n(vec![1], 2);
+    add_n_inplace(&mut v, 2);
+    dedup(&mut v);
+    println!("{:?}", v);
 }
 
 fn add_n(v: Vec<i32>, n: i32) -> Vec<i32> {
-    unimplemented!()
+    let mut r = Vec::new();
+    for e in v.iter() {
+        r.push(e+n);
+    };
+    r
 }
 
 fn add_n_inplace(v: &mut Vec<i32>, n: i32) {
-    unimplemented!()
+    for i in 0..v.len() {
+        v[i] += n;
+    }
 }
 
 fn dedup(v: &mut Vec<i32>) {
-    unimplemented!()
+    let mut elements = HashSet::new();
+    let mut i = 0;
+    while let Some(e) = v.get(i){
+        if elements.contains(e) {
+            v.remove(i);
+            continue;
+        }
+        elements.insert(*e);
+        i += 1;
```

Comment:
I have a doubt here, if i m insertingg `*e` then why m I not able to do `elements.contains(*e)`